src/ucp/core/ucp_mm.c

```diff
@@ -792,16 +792,17 @@ ucs_status_t ucp_memh_get_slow(ucp_context_h context, void *address,
         status = ucp_memh_rcache_get(context->rcache, reg_address, reg_length,
                                      reg_align, mem_type, reg_md_map, uct_flags,
                                      alloc_name, &memh);
-
-        ucs_assert(memh->mem_type == mem_type);
-        ucs_assert(ucs_padding((intptr_t)ucp_memh_address(memh), reg_align) == 0);
-        ucs_assert(ucs_padding(ucp_memh_length(memh), reg_align) == 0);
     }
 
     if (status != UCS_OK) {
         goto out;
     }
 
+    ucs_assert(memh->mem_type == mem_type);
+    ucs_assert((context->rcache) == NULL ||
```

Comment:
maybe put these assertions under `if (context->rcache == NULL) {`?
also may use this opportunity to convert them into ucs_assertv

Reply:
up to now we've tucked the assertion logic right into the assertions themselves
if you're thinking of switching this up for new code, let's give everyone a heads-up
otherwise you'll have to bring it up each time, as we've see here and [there](https://github.com/openucx/ucx/pull/9449#discussion_r1509907227)

Reply:
it depends on the assertion condition itself. in this particular case, putting the rcache!=NULL check inside the assertion logic makes it more complicated, unites 2 assertions to one (so when it fails, it's harder to know which one failed), and the check or "rcache == NULL" is not the essence of the assertion itself, but just the condition under which we want to perform the check (otherwise, memh is not even valid)

Reply:
also use `ucs_assertv()` instead and dump all fields (length, address, align and padding)?